src/services/item/plugins/itemLike/index.ts

```diff
@@ -34,9 +38,25 @@ const plugin: FastifyPluginAsync = async (fastify) => {
   fastify.post<{ Params: { itemId: string } }>(
     '/:itemId/like',
     { schema: create, preHandler: fastify.verifyAuthentication },
-    async ({ member, params: { itemId }, log }) => {
+    async (request) => {
+      const {
+        member,
+        params: { itemId },
+      } = request;
       return db.transaction(async (manager) => {
-        return itemLikeService.post(member, buildRepositories(manager), itemId);
+        // action like item
+        const item = await itemService.get(member, buildRepositories(manager), itemId);
+        const action = {
+          item,
+          type: Triggers.ItemLike,
+          extra: {
+            itemId: item?.id,
```

Comment:
```suggestion
            itemId: item.id,
```

Reply:
Maybe let's create the action after performing the like, so that the scope of `action` is smaller.